test/e2e/receive_test.go

```diff
@@ -56,6 +56,77 @@ func ErrorHandler(_ http.ResponseWriter, _ *http.Request, err error) {
 
 func TestReceive(t *testing.T) {
 	t.Parallel()
+	// t.Run("receive_distributor_ingestor_mode", func(t *testing.T) {
```

Comment:
Why these are commented? Should we mark this PR as WIP?

Reply:
I wanted to check if the original tests pass after refactoring. I haven't added the logic for split behaviour, so once the tests pass, I will uncomment them.

Reply:
Added the tests back